x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -68,139 +41,201 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
-		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
-	}
-
-	// ensure that we always have 2 tokens
-	// Note: tokens returns a random subset of poolDenoms, so  had to add this assertion
-	if len(tokens) < 2 {
-		return nil, fmt.Errorf("user doesnot have pool tokens")
-	}
-
-	//  Retrieve minTick and maxTick from precision factor
-	minTick, maxTick := clkeeper.GetMinAndMaxTicksFromExponentAtPriceOne(clPool.GetExponentAtPriceOne())
-
-	// Randomize lowerTick and upperTick from max values to create position
-	lowerTick, upperTick, err := getRandomTickPositions(sim, minTick, maxTick, clPool.GetTickSpacing())
+	positionCreator, tokens, lowerTick, upperTick, err := RandomPrepareCreatePositionFunc(sim, ctx, clPool, poolDenoms)
 	if err != nil {
 		return nil, err
 	}
 
 	return &cltypes.MsgCreatePosition{
-		PoolId:        clPool.GetId(),
-		Sender:        sender.Address.String(),
-		LowerTick:     lowerTick,
-		UpperTick:     upperTick,
-		TokenDesired0: tokens[0],
-		TokenDesired1: tokens[1],
-		// TODO: Randomize TokenMinAmount0 and TokenMinAmount1 in next iteration
+		PoolId:          clPool.GetId(),
+		Sender:          positionCreator.String(),
+		LowerTick:       lowerTick,
+		UpperTick:       upperTick,
+		TokenDesired0:   tokens[0],
+		TokenDesired1:   tokens[1],
 		TokenMinAmount0: sdk.NewInt(0),
 		TokenMinAmount1: sdk.NewInt(0),
 	}, nil
 }
 
 func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
-	rand := sim.GetRand()
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, _, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// get random user address with the pool denoms
-	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
-	if !senderExists {
-		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	// Utilize the PoolId to PositionId mapping
+	positionIds, err := k.GetAllPositionIdsForPoolId(ctx, clPool.GetId())
+	if err != nil {
+		return nil, err
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
-	if err != nil {
-		return nil, fmt.Errorf("position does not exist")
+	if len(positionIds) < 1 {
+		return nil, fmt.Errorf("user doesnot have any positions")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	randPositionId := positionIds[rand.Intn(len(positionIds))]
+	position, err := k.GetPosition(ctx, randPositionId)
+	if err != nil {
+		return nil, err
 	}
 
-	// pick a random position
-	randPosition := positions[rand.Intn(len(positions))]
+	withdrawAmount := sim.RandomDecAmount(position.Liquidity)
 
-	// get percentage amount from 1 to 100 to withdraw liquidity
-	randPerc := sim.RandomDecAmount(sdk.OneDec())
+	// check if there is enough position liquidity to withdraw
+	availableLiquidity, err := k.GetPositionLiquidity(ctx, randPositionId)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This check isn't necessary right? We are pulling the liquidity directly from the position struct and using that as the max for the withdrawAmount

Reply:
yea this was just an extra check i added, i'll rerun the sim to see if this has any impact. And if not i'll remove it